packages/primitives/radio/src/Root.tsx

```diff
@@ -13,44 +13,56 @@ const styles = normalizeWebStyle({
   },
 })
 
+const invisibleStyles = normalizeWebStyle({
+  opacity: 0,
+  position: 'absolute',
+  top: 0,
+  zIndex: -1,
+})
+
 export const RadioInput = component(
   startWithType<TRadioInput>(),
   mapDefaultProps({
     isDisabled: false,
     accessibilityLabelBy: [],
+    isVisible: false,
   }),
-  mapContext(RadioContext)
+  mapContext(RadioContext),
+  mapHandlers({
+    onChange: ({ setGroupValue, value }) => () => setGroupValue(value),
+  })
 )(({
   id,
+  children,
   groupName,
   groupValue,
-  setGroupValue,
-  key,
+  onChange,
   accessibilityLabelBy,
   accessibilityLabel,
   value,
   isDisabled,
-  onChange,
-}) => (
-  <input
-    type="radio"
-    id={id}
-    name={groupName}
-    key={key || groupName + id}
-    checked={groupValue === value}
-    value={value}
-    aria-labelledby={accessibilityLabelBy.length > 0 ? accessibilityLabelBy.join(' ') : undefined}
-    aria-label={accessibilityLabel}
-    disabled={isDisabled}
-    style={styles}
-    onChange={(evt) => {
-      setGroupValue(evt.currentTarget.value)
+  isVisible,
+}) => {
+  const labelledBy = accessibilityLabelBy.length > 0 ? accessibilityLabelBy.join(' ') : undefined
```

Comment:
This is a debatable line. 
On one side, it is same job (joining same array) every frame.
On the other - caching it includes some prop change logic, which might be comparable in performance.